src/main/ApplicationImpl.cpp

```diff
@@ -190,105 +190,112 @@ maybeRebuildLedger(Application& app, bool applyBuckets)
     std::set<LedgerEntryType> toRebuild;
     auto& ps = app.getPersistentState();
     auto bucketListDBEnabled = app.getConfig().isUsingBucketListDB();
-    for (auto let : xdr::xdr_traits<LedgerEntryType>::enum_values())
+
+#ifdef BUILD_TESTS
+    if (app.getConfig().MODE_USES_IN_MEMORY_LEDGER)
     {
-        // If BucketListDB is enabled, drop all tables except for offers
-        LedgerEntryType t = static_cast<LedgerEntryType>(let);
-        if (let != OFFER && bucketListDBEnabled)
+        // in-memory mode must always rebuild SQL table
+        toRebuild.emplace(OFFER);
+    }
+    else
+#endif
+    {
+        for (auto let : xdr::xdr_traits<LedgerEntryType>::enum_values())
         {
-            toDrop.emplace(t);
-            continue;
-        }
+            // If BucketListDB is enabled, drop all tables except for offers
```

Comment:
I know it was here before, but shouldn't it be the other way around? Only rebuild offers, because those are in SQL? the rebuild functionality was added to safely deal with SQL schema upgrades, so only offers are impacted

Reply:
If a recall correctly this is a hacky workaround given that enabling BucketListDB is a psuedo schema upgrade, but because it's optional and based on a config flag, couldn't be a proper schema upgrade. I'd like to leave this as is here because it works, and the follow up SQL removal PR addresses this.

Reply:
well, the whole thing would still work if you removed the rebuild logic completely. Right now the logic here doesn't quite make sense: we're rebuilding everything except offers (but there's nothing to rebuild but offers because all other tables are dropped already in favor of bucketlistDB). The point of the rebuild is to guard against buggy schema upgrades (something we got bitten by in the past which super hard to recover from). I agree this is outside of the scope of this PR, but could we open an issue to fix this?

Reply:
wdyt?

Reply:
This is going away in the next PR, which I'll open right after this gets merged :)

Reply:
sounds good, thanks! just be careful and remember to still rebuild offers given that we use sql for them still.